sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/UseRowMapper.java

```diff
@@ -13,23 +13,49 @@
  */
 package org.jdbi.v3.sqlobject.statement;
 
+import java.lang.annotation.Annotation;
 import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
+import java.lang.reflect.Method;
 
 import org.jdbi.v3.core.mapper.RowMapper;
+import org.jdbi.v3.core.result.ResultSetIterable;
+import org.jdbi.v3.core.statement.UnableToCreateStatementException;
+import org.jdbi.v3.sqlobject.customizer.SqlStatementCustomizer;
+import org.jdbi.v3.sqlobject.customizer.SqlStatementCustomizerFactory;
+import org.jdbi.v3.sqlobject.customizer.SqlStatementCustomizingAnnotation;
 
 /**
  * Used to specify specific row mapper on a query method.
  */
 @Retention(RetentionPolicy.RUNTIME)
 @Target({ElementType.METHOD})
+@SqlStatementCustomizingAnnotation(UseRowMapper.Factory.class)
 public @interface UseRowMapper
 {
     /**
      * The class implementing {@link RowMapper}. It must have a no-arg constructor.
      * @return the class of row mapper to use.
      */
     Class<? extends RowMapper<?>> value();
+
+    class Factory implements SqlStatementCustomizerFactory
```

Comment:
Two things:

First, I love making this a meta-annotation that takes care of itself instead of handlers having to explicitly check fro it.

Second, I think this wants to be a `ConfigurerFactory`. for the same reasons `@MapTo` does.